core/src/main/scala/kafka/tools/StorageTool.scala

```diff
@@ -102,6 +111,23 @@ object StorageTool extends Logging {
       setClusterId(clusterId).
       setNodeId(config.nodeId).
       build()
+    val standaloneMode = namespace.getBoolean("standalone")
+    var advertisedListenerEndpoints: collection.Seq[kafka.cluster.EndPoint] = List()
+
+    val controllersQuorumVoters = namespace.getString("controller_quorum_voters")
+    if(standaloneMode && controllersQuorumVoters != null) {
+      throw new TerseFailure("Both --standalone and --controller-quorum-voters were set. Only one of the two flags can be set.")
+    }
+
+    if (standaloneMode) {
+      advertisedListenerEndpoints = config.effectiveAdvertisedBrokerListeners
+    } else if(controllersQuorumVoters != null) {
+      if (!validateControllerQuorumVoters(controllersQuorumVoters)) {
+        throw new TerseFailure("Expected schema for --controller-quorum-voters is <replica-id>[-<replica-directory-id>]@<host>:<port>")
+      }
+      advertisedListenerEndpoints = config.effectiveAdvertisedControllerListeners
```

Comment:
This is not correct, right? We need to parse the string that you validated in `validateControllerQuorumVoters(controllersQuorumVoters)`.

Reply:
Fixed.